### PR TITLE
Adding IAM role definition for MIT Open

### PR DESCRIPTION
**Pull request by blarghmatey**
Added role for granting permissions on S3 buckets needed by MIT Open app

### CODE REVIEW

File: pillar/vault/roles/aws.sls [hunk 1]
```diff
@@ -125,4 +125,43 @@ vault:
...
+        {
+          "Resource": [
+            "arn:aws:s3:::odl-discussions-{{ env }}",
+            "arn:aws:s3:::odl-discussions-{{ env }}/*"          ],
+          "Action": [
+            "s3:HeadObject",
+            "s3:Get*",
+            "s3:List*",
+            "s3:PutObject",
+            "S3:DeleteObject",
```
Comment (shaidar) [inline pillar/vault/roles/aws.sls:140]: Are those actions enough or does this need more extensive ones similar to OVS?

File: pillar/vault/roles/macros.jinja [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
-        {% raw %}
         sql: >-
-          CREATE USER "{{name}}" WITH PASSWORD '{{password}}' VALID UNTIL '{{expiration}}';
-          GRANT "{{name}}" TO "bootcamp-ecommerce" WITH ADMIN OPTION;
-          GRANT "bootcamp-ecommerce" TO "{{name}}";
-          GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO "{{name}}";
-          GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO "{{name}}";
-          ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL PRIVILEGES ON TABLES TO "{{name}}";
-          ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL PRIVILEGES ON SEQUENCES TO "{{name}}";
+          {% raw %}CREATE USER "{{name}}" WITH PASSWORD '{{password}}' VALID UNTIL '{{expiration}}';{% endraw %}
```
Comment (shaidar) [inline pillar/vault/roles/macros.jinja:3]: Why the change from having the entire block using `raw` to one per line?

Comment (blarghmatey) [reply pillar/vault/roles/macros.jinja:3]: So that we can use the approle parameter in the rendered Jinja

File: pillar/vault/roles/aws.sls (hunk 1 above)
Comment (blarghmatey) [reply pillar/vault/roles/aws.sls:140]: This should be plenty. The listing for OVS is every individual S3 permission, most of which it doesn't need.

**Comment by shaidar** [comment]
👍
